File: portal_estagios-main/src/services/EmpresaService.ts
import { hash } from "bcrypt";
import { EmpresaRepository } from "../repositories/EmpresaRepository";
import AppError from "../utils/AppError";

const SALT = 8;

export class EmpresaService {
  constructor(private repo: EmpresaRepository) {}

  async criar(data: any) {
    const existe =
      await this.repo.buscarPorCnpj(
        data.cnpj
      );

    if (existe) {
      throw new AppError(
        "Empresa já cadastrada",
        400
      );
    }

    const senhaHash = await hash(
      data.senha,
      SALT
    );

    return this.repo.criar({
      ...data,
      senha: senhaHash,
    });
  }

  listar() {
    return this.repo.listar();
  }

  buscarPorId(id: number) {
    return this.repo.buscarPorId(id);
  }

  async atualizar(
    id: number,
    data: any
  ) {
    if (data.senha) {
      data.senha = await hash(
        data.senha,
        SALT
      );
    }

    return this.repo.atualizar(
      id,
      data
    );
  }

  remover(id: number) {
    return this.repo.deletar(id);
  }

  aprovar(id: number) {
    return this.repo.atualizar(id, {
      status: "APROVADA",
    });
  }

  bloquear(id: number) {
    return this.repo.atualizar(id, {
      status: "BLOQUEADA",
    });
  }
}

File: src/main/java/com/unialfa/service/EmpresaService.java
package com.unialfa.service;

import com.unialfa.dao.EmpresaDAO;
import com.unialfa.model.Empresa;

import java.util.List;

public class EmpresaService {

    private final EmpresaDAO dao = new EmpresaDAO();

    public List<Empresa> listar() {
        return dao.listar();
    }

    public void aprovar(int id) {
        dao.aprovar(id);
    }

    public void bloquear(int id) {
        dao.bloquear(id);
    }
}